pipeline/metadata/blockpage.py

```diff
@@ -1,55 +1,57 @@
 """Matcher for response pages to blockpage signatures."""
 
 import json
-import os
+import pkgutil
 import re
-from typing import Optional, Dict
-
-import apache_beam.io.filesystems as apache_filesystems
+from typing import Optional, Dict, Pattern
 
 # Signature filenames
-FALSE_POSITIVES = 'false_positive_signatures.json'
-BLOCKPAGES = 'blockpage_signatures.json'
+FALSE_POSITIVES = 'data/false_positive_signatures.json'
+BLOCKPAGES = 'data/blockpage_signatures.json'
+
+
+def _create_regex(pattern: str) -> Pattern:
+  pattern = re.escape(pattern)
```

Comment:
You need to \\-unescape the string, as specified here: https://cloud.google.com/bigquery/docs/reference/standard-sql/operators#comparison_operators

But the original `%`s should be replaced with `.*` and `_` with `.`

My guess is that the original patterns are not properly \\-escaped, so you need to validate and fix the patterns too.

Reply:
No, this ended up not being the problem. The re.escape line is already escaping most things (like `.` and spaces) correctly.

`%` is being turned correctly into `.*` in the final regex
`_` should stay as `_`. If it was operating as a special char in bq I think that was unintended. (if you look at the patterns it's clearly not intended as a regex character)

I'll make a longer comment below.

Reply:
The url-escaped `%` chars also ended up being a red herring, although they're still technically being escaped incorrectly. As long as the regex has some unique string prefix like `https://www.divo.ru/` it starts by looking for that. Since it never finds it it never gets to the expensive many-`.*` portion of the regex.

Reply:
The code and data are still broken. We need to fix both.

First, we need to precisely define what the input format is. It's currently undefined. Is it a BigQuery LIKE string? Is it a regex?
It's currently none, since it's not a regex and we are not properly escaping as LIKE, as there are multiple `_` and `%` with the wrong semantics. We must escape them if we want to declare the input is the LIKE format.
`UrlRedir=http%3A%2F%2F1337x.to%2f` should be `UrlRedir=http\\%3A\\%2F\\%2F1337x.to\\%2f` in the json file (`UrlRedir=http\%3A\%2F\%2F1337x.to\%2f` as the parsed string).

My suggestion is to replace the pattern string with a list of strings. This way you can easily generate a LIKE string by replacing all `_\%` with the escaped version for all parts and join the result with `%`.
And you can easily generate a regex by re-escaping the parts and joining with `.*`.
Otherwise you will need to do the split, unescape, replace, re-escape, join.

Another possibility is for us to come up with our own format, where we have a special, unique mark as the place to insert the `.*` or `%`. For example, insert a mark like `$BREAK$`, assuming it doesn't occur in the patterns. Then you escape the whole pattern, and replace the escaped `$BREAK$` (`\$BREAK\$` in RE, `$BREAK$` in LIKE) with the appropriate operator.

In short, the 3 options are:

1. Use a list of strings in the patterns
2. Use LIKE format and do the split-unescape-escape-join dance
3. Use the original strings, but with a unique marker you can replace (not `%`)

@ramakrishnansr this is something that must be fixed in the pattern generation too, since it's generating patterns that are misusing the special characters.

Reply:
There's a fourth option: save the file in regex format. Then you just read and use.
If we end up needing to use the patterns in BigQuery, we can call REGEXP_CONTAINS instead of LIKE.

Reply:
I think the last option makes the most sense. I can spend some time today making sure all the patterns are in regex format and push that to our assets. Then all of the problems can be avoided.

Reply:
I agree. Anyone will be able to use and contribute with the regex format.